### PR TITLE
Switching Boskos to deployment, as well as removing persistent volume

### PR DESCRIPTION
**Pull request by chaodaiG**
This change is to make our Boskos deployment consistent with upstream, almost a mirror of https://github.com/kubernetes/test-infra/pull/13594

Also copied from the PR above:
```
kubectl delete -n test-pods statefulset boskos
# wait for boskos-0 pod to be gone in test-pods ns
kubectl delete -n test-pods pv boskos-storage
kubectl apply -f ci/prow/deployments/boskos_deployment.yaml
```

Part of: #1533 

/cc @yt3liu 
/cc @sebastienvas

### CODE REVIEW

**Comment by chaodaiG** [comment]
/hold
So that this PR doesn't get merged at a wrong time

File: ci/prow/deployments/boskos_volumes.yaml [hunk 1]
```diff
@@ -1,34 +0,0 @@
...
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-# Boskos deployment for Knative Prow instance
-apiVersion: v1
-kind: PersistentVolume
```
Comment (adrcunha) [inline ci/prow/deployments/boskos_volumes.yaml]: If you're removing this file, when creating a new cluster (or recreating boskos), won't it miss the configuration of the persistent volume used by boskos?

Comment (adrcunha) [reply ci/prow/deployments/boskos_volumes.yaml]: Wait... So the new boskos doesn't need persistent storage anymore?

Comment (chaodaiG) [reply ci/prow/deployments/boskos_volumes.yaml]: yeah that's right, I have double checked with the author of the PR in k8s/test-infra, Boskos has stopped using persistent storage about a year ago, and we are already using it. So these are just like cleaning up

Comment (adrcunha) [reply ci/prow/deployments/boskos_volumes.yaml]: Awesome. Let's delete that volume then once this is in. :)

/lgtm
/approve

**Pull request merged**